sway-core/src/declaration_engine/declaration_wrapper.rs

```diff
@@ -113,6 +113,23 @@ impl ReplaceDecls for DeclarationWrapper {
     }
 }
 
+impl ReplaceFunctionImplementingType for DeclarationWrapper {
+    fn replace_implementing_type(&mut self, implementing_type: TyDeclaration) {
+        match self {
+            DeclarationWrapper::Unknown => {}
+            DeclarationWrapper::Function(decl) => decl.set_implementing_type(implementing_type),
+            DeclarationWrapper::Trait(_) => {}
+            DeclarationWrapper::TraitFn(_) => {}
+            DeclarationWrapper::ImplTrait(_) => {}
+            DeclarationWrapper::Struct(_) => {}
+            DeclarationWrapper::Storage(_) => {}
+            DeclarationWrapper::Abi(_) => {}
+            DeclarationWrapper::Constant(_) => {}
+            DeclarationWrapper::Enum(_) => {}
```

Comment:
```suggestion
            DeclarationWrapper::Function(decl) => decl.set_implementing_type(implementing_type),
            DeclarationWrapper::Unknown
            | DeclarationWrapper::Trait(_)
            | DeclarationWrapper::TraitFn(_)
            | DeclarationWrapper::ImplTrait(_)
            | DeclarationWrapper::Struct(_)
            | DeclarationWrapper::Storage(_)
            | DeclarationWrapper::Abi(_)
            | DeclarationWrapper::Constant(_)
            | DeclarationWrapper::Enum(_) => {}
```
This actually doesn't make much difference I guess, but I'll throw it up anyway.